Source/Android/jni/MainAndroid.cpp

```diff
@@ -752,6 +752,36 @@ JNIEXPORT void JNICALL Java_org_dolphinemu_dolphinemu_NativeLibrary_ChangeDisc(J
   Core::RunAsCPUThread([&path] { DVDInterface::ChangeDisc(path); });
 }
 
+JNIEXPORT jobject JNICALL Java_org_dolphinemu_dolphinemu_NativeLibrary_GetLogTypeNames(JNIEnv* env,
+                                                                                       jobject obj)
+{
+  std::map<std::string, std::string> map;
+  Common::Log::LogManager::GetInstance()->GetLogTypes(map);
+
+  jclass mapClass = env->FindClass("java/util/LinkedHashMap");
+  if (mapClass == NULL)
+  {
+    return NULL;
+  }
+  jsize map_len = static_cast<jsize>(map.size());
+  jmethodID init = env->GetMethodID(mapClass, "<init>", "(I)V");
+  jobject linkedHashMap = env->NewObject(mapClass, init, map_len);
+  jmethodID put =
+      env->GetMethodID(mapClass, "put", "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
+  for (const auto& entry : map)
+  {
+    env->CallObjectMethod(linkedHashMap, put, env->NewStringUTF(entry.first.c_str()),
+                          env->NewStringUTF(entry.second.c_str()));
```

Comment:
Please use the `ToJString` utility function instead of using `NewStringUTF` directly.